### PR TITLE
Use params in di

### LINKED ISSUE #674: Overwrite Repositoriy Services dont work
Currently the Repositories are defined like this in the service.xml
``` xml
    <service id="sulu_media.collection_repository"
        class="%sulu_media.collection_repository.class%"
        factory-service="doctrine.orm.entity_manager"
        factory-method="getRepository">
        <argument>SuluMediaBundle:Collection</argument>
    </service>
```

If I want to use for example my own Repository I need to overwrite the class Parameter and the argument which is currently not a parameter yet. To get it work we need to change it to

``` xml
    <parameter key="sulu_media.collection_entity">SuluMediaBundle:Collection</parameter>
    <!-- ... -->
    <service id="sulu_media.collection_repository"
        class="%sulu_media.collection_repository.class%"
        factory-service="doctrine.orm.entity_manager"
        factory-method="getRepository">
        <argument>%sulu_media.collection_entity%</argument>
    </service>
```

### PR DESCRIPTION
**Pull request by dantleech**
Use parameters instead of injecting class FQNs and templates directly

Fixes #674 

__Tasks:__

- [ ] gather feedback for my changes
- [x] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fxed tickets | #674
| BC Breaks     | no
| Doc           |

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@dantleech is this ready to review?

**Comment by dantleech** [comment]
On Sun, Jan 25, 2015 at 11:28:45PM -0800, Wachter Johannes wrote:
>    [1]@dantleech is this ready to review?
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/dantleech
>    2. https://github.com/sulu-cmf/sulu/pull/735#issuecomment-71422669
>    3. https://github.com/sulu-cmf/sulu/pull/735#issuecomment-71422669

Yeah

**Comment by danrot** [comment]
Shouldn't this applied to more classes? Or are these services the only ones using the DI factory?

**Comment by dantleech** [comment]
This accounts for all of the Entity FQCN injections that were not using parameters.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,10 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * ENHANCEMENT #735 [Various]         Use parameters instead of FCQN of entities in service config
     * FEATURE     #792 [ContactBundle]   added widget to show all companys of contact
-    * BUGFIX      #801 [ALL]             Removed unused clean task which is deleting the public directory when executed
-	                                 due to the symfony 2.6 changes to symlinks
+    * BUGFIX      #801 [All]             Removed unused clean task which is deleting the public directory when executed
+	                                     due to the symfony 2.6 changes to symlinks
```
Comment (danrot) [inline CHANGELOG.md]: not aligned correctly

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,10 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * ENHANCEMENT #735 [Various]         Use parameters instead of FCQN of entities in service config
```
Comment (danrot) [inline CHANGELOG.md]: Please add a single line for every bundle

File: src/Sulu/Bundle/TagBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -17,7 +18,7 @@
         </service>
         <service id="sulu_tag.tag_repository" class="%sulu_tag.tag_repository.class%"
                  factory-method="getRepository" factory-service="doctrine.orm.entity_manager">
-            <argument>SuluTagBundle:Tag</argument>
+             <argument>%sulu_tag.entity.tag%</argument>
```
Comment (danrot) [inline src/Sulu/Bundle/TagBundle/Resources/config/services.xml]: wrong indentation

**Comment by danrot** [comment]
Otherwise ACK, if tests pass

**Pull request merged**